packages/api/src/core/prepareParams.ts

```diff
@@ -153,9 +157,30 @@ export default async function prepareParams(operation: Operation, body?: unknown
   metadata = removeUndefinedObjects(metadata);
 
   if (!jsonSchema && (body !== undefined || metadata !== undefined)) {
-    throw new Error(
-      "You supplied metadata and/or body data for this operation but it doesn't have any documented parameters or request payloads. If you think this is an error please contact support for the API you're using."
-    );
+    let throwNoParamsError = true;
+
+    // If this operation doesn't have any parameters for us to transform to JSON Schema but they've
+    // sent us either an `Accept` or `Authorization` header (or both) we should let them do that.
+    // We should, however, only do this check for the `body` parameter as if they've sent this
+    // request both `body` and `metadata` we can reject it outright as the operation won't have any
+    // body data.
+    if (body !== undefined) {
+      if (typeof body === 'object' && body !== null && !Array.isArray(body)) {
+        if (Object.keys(body).length <= 2) {
+          const bodyParams = caseless(body);
+
+          if (bodyParams.has('authorization') || bodyParams.has('accept')) {
```

Comment:
How about using your `specialHeaders` here? I just wrote this off the top of my head so it might require a bit of cleaning up

```suggestion
          if (specialHeaders.some(header => bodyParams.has(header))) {
```

Reply:
oh brilliant, i was trying to think of a cleaner way to handle this last night.